system/lib/pthread/cxa_thread_atexit.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ * Inspired by libcxxabi/src/cxa_thread_atexit.cpp.
+ * The main reasons we don't use that version direclty are:
+ * 1. We want to be able to use __cxa_thread_atexit in pure C programs
+ *    where libcxxabi is not linked in at all.
+ * 2. The libcxxabi relies on TLS variables and we use to free our TLS data
+ *    block on thread exit.  This would cause a chicken and egg issue
+ *    where the TLS variables in libcxxabi/src/cxa_thread_atexit.cpp would
+ *    be freed while porcessing the list of __cxa_thread_atexit handlers
+ */
+#include <assert.h>
+#include <stdbool.h>
+#include <stdatomic.h>
+#include <pthread.h>
+#include <stdlib.h>
+
+#include "libc.h"
```

Comment:
The includes could be:
```suggestion
#include <threads.h>

#include "libc.h"
#include "pthread_impl.h"
```
If using a `thread_local` approach (see below).